From 1a00f50e8a08c6dc0df4d8a0dfb0af29d5ca5800 Mon Sep 17 00:00:00 2001
From: Dylan Yip <dylan.yip@xilinx.com>
Date: Tue, 9 Mar 2021 12:36:43 -0800
Subject: [PATCH 1771/1851] drm: xlnx: zynqmp_disp: Fix live format config

commit 6867a2df43085fb2bedcc5db44cebb99d0a8e20a from
https://github.com/Xilinx/linux-xlnx.git

Currently the live format enums are not properly shifted. So, shift the
live format enum left by 4 as per the AV_BUF_LIVE_VID_CONFIG register
details.

While at it, use bitwise OR instead of logical OR for the display
format.

Signed-off-by: Dylan Yip <dylan.yip@xilinx.com>
Signed-off-by: Jianqiang Chen <jianqiang.chen@xilinx.com>
Acked-by: Varunkumar Allagadapa <varunkumar.allagadapa@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/gpu/drm/xlnx/zynqmp_disp.c | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/drivers/gpu/drm/xlnx/zynqmp_disp.c b/drivers/gpu/drm/xlnx/zynqmp_disp.c
index a8a4ba1e9740..ffd61635f055 100644
--- a/drivers/gpu/drm/xlnx/zynqmp_disp.c
+++ b/drivers/gpu/drm/xlnx/zynqmp_disp.c
@@ -276,10 +276,10 @@ static const u32 zynqmp_disp_gfx_init_fmts[] = {
 #define ZYNQMP_DISP_AV_BUF_LIVE_CONFIG_BPC_10		0x2
 #define ZYNQMP_DISP_AV_BUF_LIVE_CONFIG_BPC_12		0x3
 #define ZYNQMP_DISP_AV_BUF_LIVE_CONFIG_BPC_MASK		GENMASK(2, 0)
-#define ZYNQMP_DISP_AV_BUF_LIVE_CONFIG_FMT_RGB		0x0
-#define ZYNQMP_DISP_AV_BUF_LIVE_CONFIG_FMT_YUV444	0x1
-#define ZYNQMP_DISP_AV_BUF_LIVE_CONFIG_FMT_YUV422	0x2
-#define ZYNQMP_DISP_AV_BUF_LIVE_CONFIG_FMT_YONLY	0x3
+#define ZYNQMP_DISP_AV_BUF_LIVE_CONFIG_FMT_RGB		0x00
+#define ZYNQMP_DISP_AV_BUF_LIVE_CONFIG_FMT_YUV444	0x10
+#define ZYNQMP_DISP_AV_BUF_LIVE_CONFIG_FMT_YUV422	0x20
+#define ZYNQMP_DISP_AV_BUF_LIVE_CONFIG_FMT_YONLY	0x30
 #define ZYNQMP_DISP_AV_BUF_LIVE_CONFIG_FMT_MASK		GENMASK(5, 4)
 #define ZYNQMP_DISP_AV_BUF_LIVE_CONFIG_CB_FIRST		BIT(8)
 #define ZYNQMP_DISP_AV_BUF_PALETTE_MEMORY		0x400
@@ -1192,7 +1192,7 @@ static const struct zynqmp_disp_fmt av_buf_gfx_fmts[] = {
 static const struct zynqmp_disp_fmt av_buf_live_fmts[] = {
 	{
 		.bus_fmt	= MEDIA_BUS_FMT_RGB666_1X18,
-		.disp_fmt	= ZYNQMP_DISP_AV_BUF_LIVE_CONFIG_BPC_6 ||
+		.disp_fmt	= ZYNQMP_DISP_AV_BUF_LIVE_CONFIG_BPC_6 |
 				  ZYNQMP_DISP_AV_BUF_LIVE_CONFIG_FMT_RGB,
 		.rgb		= true,
 		.swap		= false,
@@ -1202,7 +1202,7 @@ static const struct zynqmp_disp_fmt av_buf_live_fmts[] = {
 		.sf[2]		= ZYNQMP_DISP_AV_BUF_6BIT_SF,
 	}, {
 		.bus_fmt	= MEDIA_BUS_FMT_RBG888_1X24,
-		.disp_fmt	= ZYNQMP_DISP_AV_BUF_LIVE_CONFIG_BPC_8 ||
+		.disp_fmt	= ZYNQMP_DISP_AV_BUF_LIVE_CONFIG_BPC_8 |
 				  ZYNQMP_DISP_AV_BUF_LIVE_CONFIG_FMT_RGB,
 		.rgb		= true,
 		.swap		= false,
@@ -1212,7 +1212,7 @@ static const struct zynqmp_disp_fmt av_buf_live_fmts[] = {
 		.sf[2]		= ZYNQMP_DISP_AV_BUF_8BIT_SF,
 	}, {
 		.bus_fmt	= MEDIA_BUS_FMT_UYVY8_1X16,
-		.disp_fmt	= ZYNQMP_DISP_AV_BUF_LIVE_CONFIG_BPC_8 ||
+		.disp_fmt	= ZYNQMP_DISP_AV_BUF_LIVE_CONFIG_BPC_8 |
 				  ZYNQMP_DISP_AV_BUF_LIVE_CONFIG_FMT_YUV422,
 		.rgb		= false,
 		.swap		= false,
@@ -1222,7 +1222,7 @@ static const struct zynqmp_disp_fmt av_buf_live_fmts[] = {
 		.sf[2]		= ZYNQMP_DISP_AV_BUF_8BIT_SF,
 	}, {
 		.bus_fmt	= MEDIA_BUS_FMT_VUY8_1X24,
-		.disp_fmt	= ZYNQMP_DISP_AV_BUF_LIVE_CONFIG_BPC_8 ||
+		.disp_fmt	= ZYNQMP_DISP_AV_BUF_LIVE_CONFIG_BPC_8 |
 				  ZYNQMP_DISP_AV_BUF_LIVE_CONFIG_FMT_YUV444,
 		.rgb		= false,
 		.swap		= false,
@@ -1232,7 +1232,7 @@ static const struct zynqmp_disp_fmt av_buf_live_fmts[] = {
 		.sf[2]		= ZYNQMP_DISP_AV_BUF_8BIT_SF,
 	}, {
 		.bus_fmt	= MEDIA_BUS_FMT_UYVY10_1X20,
-		.disp_fmt	= ZYNQMP_DISP_AV_BUF_LIVE_CONFIG_BPC_10 ||
+		.disp_fmt	= ZYNQMP_DISP_AV_BUF_LIVE_CONFIG_BPC_10 |
 				  ZYNQMP_DISP_AV_BUF_LIVE_CONFIG_FMT_YUV422,
 		.rgb		= false,
 		.swap		= false,
-- 
2.31.1

